### PR TITLE
fix(cc-widgets): ship-all-widgets-together

### PR DESCRIPTION
**Pull request by Shreyas281299**
# This PR completes [SPARK-580563](https://jira-eng-gpk2.cisco.com/jira/browse/SPARK-580563)

## What does this PR adds?
- This PR adds a new package called cc-widgets which imports and exports all the existing widgets. This will enhance the developer experience while using the widgets. Rather than adding all widgets as a dependency now developers only need to ad cc-widgets as a dependency and they will have access to all the widgets. Tree shaking will take care of removing all the unnecessary widgets that are unused in the application and the bundle size willl not increase.

## Testing performed 
- Published the newly created package onto my private npm account and used this to run the widgets. (bundle size for this app came around 5MB, this uses a react app and I didnt minimfy the bundle either)
- Unit testing.

## Summary by CodeRabbit

- **New Features**
	- Introduced the `@webex/cc-widgets` package with build and testing capabilities.
	- Added new imports for `StationLogin`, `UserState`, and `store` to enhance module functionality.

- **Bug Fixes**
	- Improved error handling in the `publish.js` file for better clarity and streamlined control flow.

- **Documentation**
	- Updated test suite for the `versionAndPublish` function to enhance clarity and robustness.

- **Chores**
	- Added Babel and TypeScript configuration files to ensure consistent behavior across the project.

### CODE REVIEW

File: packages/contact-center/cc-widgets/package.json
Comment (Shreyas281299) [inline packages/contact-center/cc-widgets/package.json:4]: This will get updated with the latest version by semantic-release once we merge the PR.

File: packages/contact-center/store/package.json
Comment (Shreyas281299) [inline packages/contact-center/store/package.json:21]: Using only @webex/plugin-cc rather than the whole webex object brought down the bundle size of store from 17MB to 11 MB.

File: webpack.config.js
Comment (Shreyas281299) [inline webpack.config.js:17]: Was getting fallback errors for these 2 as wel.

File: tooling/src/publish.js
Comment (Shreyas281299) [inline tooling/src/publish.js:58]: We just need to update all the versions for each package.

File: tooling/src/publish.js
Comment (Shreyas281299) [inline tooling/src/publish.js:61]: Once we have all the UPDATED versions in the package.json when we publish the npm publish command takes care of replacing the `workspace:*` with the actual version from the package.json

**Comment by bhabalan** [comment]
@coderabbitai review

File: tooling/src/publish.js
Comment (Shreyas281299) [reply tooling/src/publish.js:54]: - Both the version and publish are already inside a try catch adding more try catch might be an overkill.
- The new version that we are getting is from semantic-release so we can trust that it will be in the correct format

File: tooling/src/publish.js
Comment (Shreyas281299) [reply tooling/src/publish.js:42]: - Added package.json not found error handling
- we will always have a name in the package.json data.

File: tooling/src/publish.js
Comment (Shreyas281299) [reply tooling/src/publish.js:61]: Ack.

File: tooling/src/publish.js
Comment (Shreyas281299) [reply tooling/src/publish.js:16]: Ack

File: packages/contact-center/cc-widgets/package.json
Comment (Shreyas281299) [reply packages/contact-center/cc-widgets/package.json:24]: Ack

File: packages/contact-center/cc-widgets/webpack.config.js
Comment (Shreyas281299) [reply packages/contact-center/cc-widgets/webpack.config.js:12]: We will handle reducing the bundle size and optimization in a different ticket

File: package.json
Comment (arun3528) [inline package.json:43]: i think you can use the workspace foreach command rather then writing all individually, you can run them in parallel as well

File: packages/contact-center/cc-widgets/.babelrc.js
Comment (arun3528) [inline packages/contact-center/cc-widgets/.babelrc.js:1]: sorry to ask it again , i remember asking this , do we need a babel because we are consuming sdk or its for something else

File: packages/contact-center/store/package.json
Comment (arun3528) [inline packages/contact-center/store/package.json:25]: mark other packages as private

File: packages/contact-center/cc-widgets/.babelrc.js
Comment (Shreyas281299) [reply packages/contact-center/cc-widgets/.babelrc.js:1]: Jest is what is using babel config. When the tests are written in typescript I think thats when we require babel.

**Review by rsarika — APPROVED** [review]

**Review by mkesavan13 — CHANGES_REQUESTED** [review]
On a first look it looks good to me. Just a few minor things,

- We should update the sample app to use cc-widgets package
- For WC, let us write the r2wc wrapper in the cc-widgets package itself and have the import statement like `import '@webex/cc-widgets/wc' which will export and define all the WC components

File: tooling/src/publish.js
Comment (Kesari3008) [inline tooling/src/publish.js:43]: I see console logs in this file, we would remove those before merge right

Comment (Shreyas281299) [reply tooling/src/publish.js:43]: These logs for debugging the publish step. They will only appear in the circleCI

File: package.json
Comment (Shreyas281299) [reply package.json:43]: Updated the command. We can build station-login and user-state in parallel. But since cc-widgets depends on the later packages it needs to be build after them.

File: packages/contact-center/store/package.json
Comment (Shreyas281299) [reply packages/contact-center/store/package.json:25]: We cant mark them private as there are dependencies in the cc-widgets, even if I keep them as devDependencies there packages are installed. So we would need to publish these as well. Another reason for not marking them private is that it would help us in giving out separate cdns for each widgets.

File: package.json
Comment (Shreyas281299) [reply package.json:43]: We can run them in paralle. used the --topological flag which ensures that dependencies are build in a parallel first and then the cc-widgets.

File: packages/contact-center/store/package.json
Comment (Shreyas281299) [reply packages/contact-center/store/package.json:21]: After making this changes webex is not getting initialised properly and we get an error this.cc is undefined. Since we already have a ticket for optimizing the bundle sizes I will add this as part of thaat ticket as well.

File: docs/web-component-samples/src/index.ts
Comment (mkesavan13) [inline docs/web-component-samples/src/index.ts:1]: As discussed during sitdown review, let us implement this one,

```suggestion
import {store} from '@webex/cc-widgets';
import '@webex/cc-widgets/wc';
```

File: package.json
Comment (mkesavan13) [inline package.json:39]: Let us exclude the samples and let for each handle all the other packages

File: packages/contact-center/cc-widgets/package.json
Comment (mkesavan13) [inline packages/contact-center/cc-widgets/package.json:30]: Let's remove it from `devDependencies`

Comment (Shreyas281299) [reply packages/contact-center/cc-widgets/package.json:34]: Removed

File: docs/web-component-samples/src/index.ts
Comment (Shreyas281299) [reply docs/web-component-samples/src/index.ts]: Done

File: package.json
Comment (Shreyas281299) [reply package.json]: Done

File: docs/web-component-samples/tsconfig.json
Comment (Shreyas281299) [inline docs/web-component-samples/tsconfig.json:8]: Little context- https://www.typescriptlang.org/docs/handbook/release-notes/typescript-4-7.html
When we are using export in package.json we need smarter way of compilling so that it picks up the types.

File: package.json
Comment (Shreyas281299) [inline package.json:39]: Excluding webex-widgets, as the tests are passing but it's timing out and failling the build

File: docs/web-component-samples/webpack.config.js
Comment (Shreyas281299) [inline docs/web-component-samples/webpack.config.js:7]: redundant

File: package.json
Comment (Shreyas281299) [inline package.json:38]: this should be test:unit

File: packages/contact-center/station-login/package.json
Comment (Shreyas281299) [inline packages/contact-center/station-login/package.json:54]: Remove this, was testing

**Review by mkesavan13 — APPROVED** [review]
Had a sit down review and added a few comments. Otherwise looks good to me. Approving

File: package.json
Comment (Shreyas281299) [reply package.json:38]: Updated

File: package.json
Comment (Shreyas281299) [reply package.json]: Update: webex-widgets is working fine the failure was because no tests found in cc-widgets, added passWithNoTest flag in jest config

Comment (Shreyas281299) [reply package.json]: UPDATE!! There are no unit tests in widgets

**Pull request merged**
